From bbff63f86513d43c70649ad3d2adc57254e821cd Mon Sep 17 00:00:00 2001
From: "Artyom V. Poptsov" <poptsov.artyom@gmail.com>
Date: Tue, 24 Sep 2024 20:13:23 +0300
Subject: [PATCH] examples/ssshd.scm.in (main): Bugfix: Handle deprecation of
 DSA

* examples/ssshd.scm.in (main): Bugfix: Don't try to use DSA keys when DSA is
not supported.
---
 examples/ssshd.scm.in | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/examples/ssshd.scm.in b/examples/ssshd.scm.in
index 143317c..b7bfbc9 100644
--- a/examples/ssshd.scm.in
+++ b/examples/ssshd.scm.in
@@ -5,7 +5,7 @@
 
 ;;; ssshd.scm -- Scheme Secure Shell Daemon.
 
-;; Copyright (C) 2013-2021 Artyom V. Poptsov <poptsov.artyom@gmail.com>
+;; Copyright (C) 2013-2024 Artyom V. Poptsov <poptsov.artyom@gmail.com>
 ;;
 ;; This program is free software: you can redistribute it and/or
 ;; modify it under the terms of the GNU General Public License as
@@ -38,7 +38,8 @@
              (ssh session)
              (ssh channel)
              (ssh key)
-             (ssh auth))                ; userauth-*
+             (ssh auth)                 ; userauth-*
+             (ssh version))
 
 
 ;;; Variables and constants
@@ -280,7 +281,7 @@ Options:
 
     (let ((server (make-server #:bindport      bindport
                                #:rsakey        rsakey
-                               #:dsakey        dsakey
+                               #:dsakey        (and (dsa-support?) dsakey)
                                #:log-verbosity (string->symbol ssh-debug)
                                #:banner        "Scheme Secure Shell Daemon"))
           (channel #f))
